airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -116,10 +118,18 @@ public Optional<Long> createResetConnectionJob(final DestinationConnection desti
             workerResourceRequirements))
         .withResetSourceConfiguration(new ResetSourceConfiguration().withStreamsToReset(streamsToReset));
 
+    getCurrentConnectionState(standardSync.getConnectionId()).ifPresent(resetConnectionConfig::withState);
+
     final JobConfig jobConfig = new JobConfig()
         .withConfigType(ConfigType.RESET_CONNECTION)
         .withResetConnection(resetConnectionConfig);
     return jobPersistence.enqueueJob(standardSync.getConnectionId().toString(), jobConfig);
   }
 
+  // TODO (https://github.com/airbytehq/airbyte/issues/13620): update this method implementation
+  // to fetch and serialize the new per-stream state format into a State object
+  private Optional<State> getCurrentConnectionState(final UUID connectionId) throws IOException {
```

Comment:
Ah...I see.  Disregard my previous comment unless you also want to pass the function in here and return void.